core/lib/basic_types/src/url.rs

```diff
@@ -0,0 +1,117 @@
+use std::{fmt, str::FromStr};
+
+use serde::{Deserialize, Deserializer};
+use url::Url;
+
+/// URL with potentially sensitive authentication info (user name and password). Has a specialized
+/// `Debug` implementation to hide sensitive parts and explicit API to expose the underlying URL.
+#[derive(Clone, PartialEq)]
+pub struct SensitiveUrl {
```

Comment:
That's neat, although I somewhat feel like it may be overly specific. I see that we already have [`K256PrivateKey`](https://github.com/matter-labs/zksync-era/blob/main/core/lib/crypto_primitives/src/ecdsa_signature.rs#L47) that generally serves the similar purpose.

At this stage, wouldn't it be better to just use [secrecy](https://docs.rs/secrecy/latest/secrecy/) (I feel that both this PR and the interface of `K256PrivateKey` are kinda influenced by it)?

Reply:
Yeah, I was inspired by `secrecy` when naming method. IMO, it's sort of doesn't work in either case:

- Private keys from libsecp256k1 already implement most of necessary functionality (zeroize on drop, opaque `Debug`); it's just that their `Debug` impl doesn't work the best in our case (since it wasn't designed with Ethereum in mind).
- For URLs, we'd arguably want to output *some* parts of an URL in `Debug`; hence, a custom type.